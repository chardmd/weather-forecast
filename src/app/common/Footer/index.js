import styled from 'styled-components/macro';

const Footer = () => {
  return (
    <Wrapper>
      <Span1>Designed and Handcrafted by</Span1>
      <Link href="https://chardmd.com" target="_blank">
        Richard
      </Link>
      <Span2>ðŸ¥³</Span2>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  margin: 0 auto;
  padding: 1rem;
  background-color: ${p => p.theme.backgroundVariant};
  color: ${p => p.theme.text};
  bottom: 0;
  left: 0;
  width: 100%;
  min-width: 250px;
  text-align: center;
  position: fixed;
`;

const Link = styled.a`
  color: ${p => p.theme.highLight};
`;

const Span1 = styled.span`
  margin-right: 0.5rem;
`;

const Span2 = styled.span`
  margin-left: 0.5rem;
`;

export default Footer;
