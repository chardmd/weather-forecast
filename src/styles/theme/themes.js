const mongoDBTheme = {
  primary: '#4DB33D',
  text: '#589636',
  textSecondary: '#3F3E42',
  background: '##E8E7D5',
  backgroundVariant: '#E8E7D5',
  border: '#3F3E42',
  borderLight: '#C1BEBC',
};

export const themes = {
  mongoDBTheme: mongoDBTheme,
};
