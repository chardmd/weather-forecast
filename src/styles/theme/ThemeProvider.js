import * as React from 'react';
import { ThemeProvider as OriginalThemeProvider } from 'styled-components';
import { themes } from './themes';

export const ThemeProvider = props => {
  return (
    <OriginalThemeProvider theme={themes.mongoDBTheme}>
      {props.children}
    </OriginalThemeProvider>
  );
};
