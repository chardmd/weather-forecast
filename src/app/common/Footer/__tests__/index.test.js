import * as React from 'react';
import { render } from '@testing-library/react';

import Footer from '../index';
import { themes } from '../../../../styles/theme/themes';
import { ThemeProvider } from 'styled-components';

const renderWithTheme = (props, theme) =>
  render(
    <ThemeProvider theme={theme || themes.mongoInspiredTheme}>
      <Footer {...props} />
    </ThemeProvider>,
  );

describe('<Footer />', () => {
  it('should match snapshot', () => {
    const component = renderWithTheme();
    expect(component.container.firstChild).toMatchSnapshot();
  });
});
