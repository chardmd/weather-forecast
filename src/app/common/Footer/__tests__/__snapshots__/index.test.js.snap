// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
.c0 {
  margin: 0 auto;
  padding: 1rem;
  background-color: #E8E7D5;
  color: #589636;
  bottom: 0;
  left: 0;
  width: 100%;
  min-width: 250px;
  text-align: center;
  position: fixed;
}

.c2 {
  color: #1DA1F2;
}

.c1 {
  margin-right: 0.5rem;
}

.c3 {
  margin-left: 0.5rem;
}

<div
  class="c0"
>
  <span
    class="c1"
  >
    Designed and Handcrafted by
  </span>
  <a
    class="c2"
    href="https://chardmd.com"
    target="_blank"
  >
    Richard
  </a>
  <span
    class="c3"
  >
    ðŸ¥³
  </span>
</div>
`;
